src/pages/instances/forms/InstanceRename.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { Button, Input } from "@canonical/react-components";
+import { renameInstance } from "api/instances";
+import { useNotify } from "context/notify";
+import { useNavigate } from "react-router-dom";
+import SubmitButton from "components/SubmitButton";
+
+interface Props {
+  instance: LxdInstance;
+  project: string;
+  closeForm: () => void;
+}
+
+const InstanceRename: FC<Props> = ({ instance, project, closeForm }) => {
+  const navigate = useNavigate();
+  const notify = useNotify();
+  const [name, setName] = useState(instance.name);
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  const handleSubmit = () => {
+    if (instance.name === name) {
+      closeForm();
+      return;
+    }
+    setSubmitting(true);
+    renameInstance(instance.name, name, project)
+      .then(() => {
+        navigate(
+          `/ui/${project}/instances/detail/${name}`,
+          notify.queue(notify.success("Instance renamed."))
+        );
+        closeForm();
+      })
+      .catch((e) => notify.failure("", e, undefined, "Renaming failed"))
```

Comment:
This should be updated according to the new order of parameters for the `failure` function, as set in #272.

Reply:
right, now that it is merged, we can adjust it as well.